src/cc_catalog_airflow/dags/provider_api_scripts/test_metropolitan_museum_of_art.py

```diff
@@ -0,0 +1,141 @@
+        mma.delayed_requester,
+        'get',
+        return_value=r
+    ) as mock_get:
+        total_objects = mma._get_object_ids('')
+    assert total_objects[0] == 4
+    assert total_objects[1] == [153, 1578, 465, 546]
+
+
+def test_get_response_json_retries_with_none_response():
+    with patch.object(
+            mma.delayed_requester,
+            'get',
+            return_value=None
+    ) as mock_get:
+        with pytest.raises(Exception):
+            assert mma._get_response_json(None, '', retries=2)
+
+    assert mock_get.call_count == 3
+
+
+def test_get_response_json_retries_with_non_ok():
+    r = requests.Response()
+    r.status_code = 504
+    r.json = MagicMock(return_value={})
+    with patch.object(
+            mma.delayed_requester,
+            'get',
+            return_value=r
+    ) as mock_get:
+        with pytest.raises(Exception):
+            assert mma._get_response_json(None, '', retries=2)
+
+    assert mock_get.call_count == 3
+
+
+def test_get_response_json_returns_response_json_when_all_ok():
+    expect_response_json = {
+        'accessionNumber': '36.100.45',
+        'artistAlphaSort': 'Kiyohara Yukinobu',
+        'artistBeginDate': '1643'
+    }
+    r = requests.Response()
+    r.status_code = 200
+    r.json = MagicMock(return_value=expect_response_json)
+    with patch.object(
+            mma.delayed_requester,
+            'get',
+            return_value=r
+    ) as mock_get:
+        actual_response_json = mma._get_response_json(None, '', retries=2)
+
+    assert mock_get.call_count == 1
+    assert actual_response_json == expect_response_json
+
+
+def test_create_meta_data():
+    exact_response = {
+        'accession_number': '36.100.45',
+        'classification': 'Paintings',
+        'credit_line': (
+            'The Howard Mansfield Collection, Purchase, Rogers Fund, 1936'
+            ),
+        'culture': 'Japan'
+    }
+    r = requests.Response()
+    r.status_code = 200
+    r.json = MagicMock(return_value=exact_response)
+    with patch.object(
+        mma.delayed_requester,
+        'get',
+        return_value=r
+    ) as mock_get:
+        response = mma._get_response_json(None, '', retries=2)
+
+    assert response == exact_response
+
+
+def test_process_image_data_handles_example_dict():
+    with open(os.path.join(RESOURCES, 'sample_image_data.json')) as f:
+        image_data = json.load(f)
+
+    with patch.object(
+        mma.image_store,
+        'add_item',
+        return_value=1
+    ) as mock_add:
+        mma._process_image_data(image_data)
+
+    mock_add.assert_called_once_with(
+        foreign_landing_url=(
+            "https://wwwstg.metmuseum.org/art/collection/search/45734"
+            ),
+        image_url=(
+            "https://images.metmuseum.org/CRDImages/as/original/DP251139.jpg"
+            ),
+        thumbnail_url=(
+            "https://images.metmuseum.org/CRDImages/as/web-large/DP251139.jpg"
+            ),
+        license="cc0",
```

Comment:
this defect in the test means it's not catching the error in the code.  There's a couple of ways to avoid such things.  The easiest is to make sure that you always copy the parameter lists from the function being mocked, rather than from your own code (this helps avoid variable name typos).

Reply:
An even more robust option is [autospeccing](https://docs.python.org/3/library/unittest.mock.html#auto-speccing), but I wouldn't require that to merge the PR (perhaps migrating all tests that mock to use autospeccing could be a future issue).